src/core/DataFile.cpp

```diff
@@ -163,6 +163,53 @@ DataFile::~DataFile()
 
 
 
+bool DataFile::validate( QString extension )
+{
+	switch( m_type )
+	{
+	case Type::SongProject:
+		if( extension == "mmp" || extension == "mmpz" )
+		{
+			return true;
+		}
+		break;
+	case Type::SongProjectTemplate:
+		if(  extension == "mpt" )
+		{
+			return true;
+		}
+		break;
+	case Type::InstrumentTrackSettings:
+		if ( extension == "xpf" || extension == "xml" )
+		{
+			return true;
+		}
+		break;
+	case Type::UnknownType:
+		if (! ( extension == "mmp" || extension == "mpt" || extension == "mmpz" ||
```

Comment:
A bit late in response to this but we can get back to the pretty select/case statements by creating a FileType enum down the road.  I think it will have an added benefit by finally allowing us to remove the `mmpz`, `mpt`, `xpf` hard coded all over the place. :+1:   Probably worth an enhancement request.